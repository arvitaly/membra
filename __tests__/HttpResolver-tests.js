"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const HttpResolver_1 = require("./../HttpResolver");
it("when fetch, should call global fetch with true params", () => __awaiter(this, void 0, void 0, function* () {
    const fetchFn = jest.fn();
    const address = "http://127.0.0.1:1337/graphql";
    const resolver = new HttpResolver_1.default({ address, fetch: fetchFn });
    fetchFn.mockImplementation(() => {
        return Promise.resolve({
            json: () => {
                return { data: { test: 15 } };
            },
        });
    });
    const result = yield resolver.fetch(`query Q1{
        viewer{
            users{
                edges{
                    node{
                        name
                    }
                }
            }
        }
    }`, { id: 1 });
    expect(fetchFn.mock.calls).toMatchSnapshot();
    expect(result).toMatchSnapshot();
}));
