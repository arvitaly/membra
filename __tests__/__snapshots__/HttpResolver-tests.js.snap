// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when fetch, should call global fetch with true params 1`] = `
Array [
  Array [
    "http://127.0.0.1:1337/graphql",
    Object {
      "body": "{\\"query\\":\\"query Q1{\\\\n        viewer{\\\\n            users{\\\\n                edges{\\\\n                    node{\\\\n                        name\\\\n                    }\\\\n                }\\\\n            }\\\\n        }\\\\n    }\\",\\"vars\\":{\\"id\\":1}}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`when fetch, should call global fetch with true params 2`] = `
Object {
  "test": 15,
}
`;
